modules/adnowBidAdapter.js

```diff
@@ -0,0 +1,83 @@
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { NATIVE, BANNER } from '../src/mediaTypes.js';
+import * as utils from '../src/utils.js';
+
+const BIDDER_CODE = 'adnow';
+const ENDPOINT = '//n.ads3-adnow.com/a';
+
+/** @type {BidderSpec} */
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [ NATIVE, BANNER ],
+
+  /**
+   * @param {object} bid
+   * @return {boolean}
+   */
+  isBidRequestValid: (bid) => {
+    return !!(bid && bid.params && (bid.params.codeId));
+  },
+
+  /**
+   * @param {BidRequest[]} bidRequests
+   * @param {*} bidderRequest
+   * @return {ServerRequest}
+   */
+  buildRequests: (bidRequests, bidderRequest) => {
+    return bidRequests.map(req => {
+      return {
+        method: 'GET',
+        url: ENDPOINT,
+        data: utils.parseQueryStringParameters({
+          Id: req.params.codeId,
+          mediaType: req.params.mediaType || 'native',
```

Comment:
Should check this against your supported media types. As it is now any string can be passed in here.